src/allencell_ml_segmenter/thresholding/thresholding_view.py

```diff
@@ -0,0 +1,343 @@
+        """
+        self._thresholding_model.set_autothresholding_enabled(
+            self._autothreshold_radio_button.isChecked()
+        )
+        self._autothreshold_method_combo.setEnabled(
+            self._autothreshold_radio_button.isChecked()
+        )
+
+        self._thresholding_model.set_threshold_enabled(
+            self._specific_value_radio_button.isChecked()
+        )
+        self._enable_specific_threshold_widgets(
+            self._specific_value_radio_button.isChecked()
+        )
+
+        self._apply_save_button.setEnabled(
+            self._specific_value_radio_button.isChecked()
+            or self._autothreshold_radio_button.isChecked()
+        )
+
+    def _check_able_to_threshold(self) -> bool:
+        able_to_threshold: bool = True
+        # Check to see if output directory is selected
+        if self._file_input_model.get_output_directory() is None:
+            show_info("Please select an output directory first.")
+            able_to_threshold = False
+
+        # Check to see if input images / directory of images are selected
+        if self._file_input_model.get_input_mode() is None:
+            show_info("Please select an input mode first.")
+            able_to_threshold = False
+        else:
+            if (
+                self._file_input_model.get_input_mode()
+                == InputMode.FROM_NAPARI_LAYERS
+                and self._file_input_model.get_selected_paths() is None
+            ):
+                show_info("Please select on screen images to threshold.")
+                able_to_threshold = False
+            elif (
+                self._file_input_model.get_input_mode() == InputMode.FROM_PATH
+                and self._file_input_model.get_input_image_path() is None
+            ):
+                show_info("Please select a directory to threshold.")
+                able_to_threshold = False
+
+        # check to see if thresholding method is selected
+        if (
+            not self._thresholding_model.is_threshold_enabled()
+            and not self._thresholding_model.is_autothresholding_enabled()
+        ):
+            show_info("Please select a thresholding method first.")
+            able_to_threshold = False
+
+        return able_to_threshold
+
+    def _save_thresholded_images(self) -> None:
+        output_dir: Optional[Path] = (
+            self._file_input_model.get_output_directory()
+        )
+        if output_dir is not None and self._check_able_to_threshold():
+            progress_tracker: PredictionFolderProgressTracker = (
+                PredictionFolderProgressTracker(
+                    output_dir,
+                    len(self._file_input_model.get_input_files_as_list()),
+                )
+            )
+
+            self.startLongTaskWithProgressBar(progress_tracker)
+
+    def doWork(self) -> None:
+        self._thresholding_model.dispatch_save_thresholded_images()
```

Comment:
for an image dir, is this when thresholding is applied, or just when disk io happens?